tests/regression/test_neg_rebase_sanity_checks.py

```diff
@@ -0,0 +1,113 @@
+import pytest
+from brownie import Contract, web3, reverts, accounts, chain  # type: ignore
+from utils.evm_script import encode_error
+from utils.test.exit_bus_data import LidoValidator
+from utils.test.extra_data import ExtraDataService
+from utils.test.oracle_report_helpers import (
+    oracle_report,
+)
+
+from utils.test.helpers import ETH
+from utils.config import (
+    contracts,
+)
+
+INITIAL_SLASHING_AMOUNT_PWEI = 1000
+INACTIVITY_PENALTIES_AMOUNT_PWEI = 101
+ONE_PWEI = ETH(0.001)
+
+@pytest.fixture(scope="module")
+def oracle_report_sanity_checker() -> Contract:
+    return contracts.oracle_report_sanity_checker
+
+
+def test_negative_rebase_correct_exited_validators_count_pos_rebase(oracle_report_sanity_checker):
+    locator = contracts.lido_locator
+    assert oracle_report_sanity_checker.address == locator.oracleReportSanityChecker()
+
+    reported_validators = exited_validators_count()
+
+    reported_validators_values = [value + 2 for value in reported_validators.values()]
+    oracle_report(cl_diff=ETH(300), stakingModuleIdsWithNewlyExitedValidators=list(reported_validators.keys()),
+                   numExitedValidatorsByStakingModule=reported_validators_values)
+
+    count = oracle_report_sanity_checker.getReportDataCount()
+    assert count > 0
+    (_, stored_exited_validators, _) = oracle_report_sanity_checker.reportData(count - 1)
+
+    assert stored_exited_validators == sum(reported_validators_values)
+
+def test_negative_rebase_correct_exited_validators_count_neg_rebase(oracle_report_sanity_checker):
+    locator = contracts.lido_locator
+    assert oracle_report_sanity_checker.address == locator.oracleReportSanityChecker()
+
+    reported_validators = exited_validators_count()
+
+    reported_validators_values = [value + 3 for value in reported_validators.values()]
+    oracle_report(cl_diff=-ETH(40000), stakingModuleIdsWithNewlyExitedValidators=list(reported_validators.keys()),
+                   numExitedValidatorsByStakingModule=reported_validators_values)
+
+    count = oracle_report_sanity_checker.getReportDataCount()
+    assert count > 0
+    (_, stored_exited_validators, _) = oracle_report_sanity_checker.reportData(count - 1)
+
+    assert stored_exited_validators == sum(reported_validators_values)
+
+def test_negative_rebase_correct_balance_neg_rebase(oracle_report_sanity_checker):
+    locator = contracts.lido_locator
+    assert oracle_report_sanity_checker.address == locator.oracleReportSanityChecker()
+
+    cl_decrease = ETH(40000)
+    oracle_report(cl_diff=-cl_decrease, exclude_vaults_balances=True)
+
+    count = oracle_report_sanity_checker.getReportDataCount()
+    assert count > 0
+    (_, _, cl_balance) = oracle_report_sanity_checker.reportData(count - 1)
+    assert cl_balance == cl_decrease
```

Comment:
I am curious why we get this result , cl_diff was -40_000 ETH then cl_balance in positive 40_000 ETH

Reply:
It's the result on how sanity checker keeps the negative rebases. It stores negative balance changes as a positive number (abs delta) or 0 if there was a positive rebase. To logic behind it that it's only negative changes are important and it tracks the cumulative negative rebases over 54 days. Here is the corresponding code https://github.com/lidofinance/core/blob/669d8e57ddcd6605ee076fe2fcdca5d4751fa154/contracts/0.8.9/sanity_checks/OracleReportSanityChecker.sol#L675